core/rfb.js

```diff
@@ -919,6 +996,23 @@
 
             if (this._sock.rQwait("sub auth capabilities", 16 * subAuthCount, 4)) { return false; }
 
+            // Newer X10 Supermicro motherboards get here.
+            // N.B.(kelleyk): If we had trouble with this heuristic matching
+            // non-ATEN servers, we could also add the "only security type 0x10
+            // is supported" condition from above.
+            if (this._rfb_version === 3.8 &&
+                    numTunnels === 0 &&
+                    (subAuthCount === 0 || (subAuthCount & 0xFFFF) == 0x0100)) {
+                Util.Info('Detected ATEN iKVM server (using heuristic #1 -- newer AST2400 BMC?).');
+                return this._negotiate_aten_auth();
+            }
+
+            // empty sub-auth list received means 'no auth' subtype selected
+            if (subAuthCount === 0) {
```

Comment:
Please drop it in that case. If it is added I'd like to have a better rationale as why this wasn't needed until now.